### ISSUE TITLE
autoscaling:DescribeAutoScalingInstances required even when checkASGTagBeforeDraining set to false

### ISSUE THREAD

**Issue by vumdao** [aws/aws-node-termination-handler#616]
**Describe the bug**
NTH is installed with `checkASGTagBeforeDraining: false` and using SQS processor mode, but when consuming message, NTH got error `not authorized to perform: autoscaling:DescribeAutoScalingInstances`

**Steps to reproduce**
- Start NTH with SQS processor mode
- Allow NTH serviceAccount permission to consume SQS message and perform EC2 actions only
- When there's event such as `EC2 Instance Rebalance Recommendation` issue happens
```
# kubectl get pod -n kube-system |grep handler
aws-node-termination-handler-5889bcd797-h2sth   0/1     CrashLoopBackOff   31         11h
```

**Expected outcome**
NTH does not need permission of `autoscaling` because we're using karpenter and no ASG at all

**Application Logs**
```
2022/04/11 03:02:52 ERR ignoring interruption event due to error error="unable to retrieve AutoScaling group: AccessDenied: User: arn:aws:sts::123456789012:assumed-role/nth-role/1649646169682969542 is 
not authorized to perform: autoscaling:DescribeAutoScalingInstances because no identity-based policy allows the autoscaling:DescribeAutoScalingInstances action\n\tstatus code: 403, request id: 6773badf-162f-41c2
-966e-6eb54fd2b32b"
2022/04/11 03:02:52 ERR error processing interruption events error="some interruption events for message Id 1100000000000000010110000011001010001000 could not be processed"
```
**Environment**

* NTH App Version: v1.16.0
* NTH Mode (IMDS/Queue processor): SQS Queue processor
* OS/Arch: Bottlerocket
* Kubernetes version: 1.20
* Installation method: Helm chart

**Comment by brycahta** [comment]
@vumdao thanks for the issue and identifying a workaround while we investigate. Taking a look

**Comment by AustinSiu** [comment]
Fix released as part of [v1.16.3](https://github.com/aws/aws-node-termination-handler/releases/tag/v1.16.3), closing the issue.
